packages/immutadot-parser/src/toPath.ts

```diff
@@ -1,128 +1,128 @@
-import {
-  allProps,
-  index,
-  list,
-  prop,
-  slice,
-} from './consts'
+import { NavType } from "./enums";
 
 import {
   filter,
   map,
+  PathParser,
   race,
-  regexp,
-} from './parser.utils'
+  regexpToParser,
+} from "./parser.utils";
 
 import {
   isIndex,
   isNil,
   isSliceIndex,
   toString,
-} from './utils'
+} from "./utils";
+
+const { allProps, index, list, prop, slice } = NavType;
 
 /**
  * Strip slashes preceding occurences of <code>quote</code> from <code>str</code><br />
  * Possible quotes are <code>"</code> and <code>'</code>.
- * @function
- * @param {string} str The string
- * @param {string} quote The quote to unescape
- * @returns {string} The unescaped string
- * @memberof path
- * @private
- * @since 1.0.0
+ *
+ * @param str The string
+ * @param quote The quote to unescape
+ * @returns The unescaped string
+ *
+ * @remarks
+ * Since 1.0.0
  */
-const unescapeQuotes = (str, quote) => str.replace(new RegExp(`\\\\${quote}`, 'g'), quote)
+const unescapeQuotes = (str: string, quote: "'" | '"') => str.replace(new RegExp(`\\\\${quote}`, "g"), quote);
 
 /**
  * Converts <code>str</code> to a slice index.
- * @function
- * @param {string} str The string to convert
- * @param {number?} defaultValue The default value if <code>str</code> is empty
- * @returns {number} <code>undefined</code> if <code>str</code> is empty, otherwise an int (may be NaN)
- * @memberof path
- * @private
- * @since 1.0.0
+ *
+ * @param str The string to convert
+ * @param defaultValue The default value if <code>str</code> is empty
+ * @returns <code>undefined</code> if <code>str</code> is empty, otherwise an int (may be NaN)
+ *
+ * @remarks
+ * Since 1.0.0
  */
-const toSliceIndex = (str, defaultValue) => str === '' ? defaultValue : Number(str)
+const toSliceIndex = (str: string, defaultValue: number = undefined) => str === "" ? defaultValue : Number(str);
```

Comment:
Again, I don't understand how this compiles. `undefined` is not a `number`... Oh! maybe you are not using the strict mode?

Reply:
I confirm this does not compile. But you can use `defaultValue?: number`

Reply:
Done